node/src/components/block_accumulator/tests.rs

```diff
@@ -1901,3 +1904,190 @@ async fn block_accumulator_reactor_flow() {
             .contains_key(old_era_block.hash()));
     }
 }
+
+#[tokio::test]
+async fn block_accumulator_doesnt_purge_with_delayed_block_execution() {
+    let mut rng = TestRng::new();
+    let (chainspec, chainspec_raw_bytes) =
+        <(Chainspec, ChainspecRawBytes)>::from_resources("local");
+    let mut runner: Runner<MockReactor> = Runner::new(
+        (),
+        Arc::new(chainspec),
+        Arc::new(chainspec_raw_bytes),
+        &mut rng,
+    )
+    .await
+    .unwrap();
+
+    // Create 1 block.
+    let block_1 = generate_non_genesis_block(&mut rng);
+
+    // Also create 2 peers.
+    let peer_1 = NodeId::random(&mut rng);
+    let peer_2 = NodeId::random(&mut rng);
+
+    let fin_sig_bob = FinalitySignature::create(
+        *block_1.hash(),
+        block_1.header().era_id(),
+        &BOB_SECRET_KEY,
+        BOB_PUBLIC_KEY.clone(),
+    );
+
+    let fin_sig_carol = FinalitySignature::create(
+        *block_1.hash(),
+        block_1.header().era_id(),
+        &CAROL_SECRET_KEY,
+        CAROL_PUBLIC_KEY.clone(),
+    );
+
+    let fin_sig_alice = FinalitySignature::create(
+        *block_1.hash(),
+        block_1.header().era_id(),
+        &ALICE_SECRET_KEY,
+        ALICE_PUBLIC_KEY.clone(),
+    );
+
+    // Register the era in the validator matrix so the block is valid.
+    {
+        let mut validator_matrix = runner.reactor_mut().validator_matrix.clone();
+        let weights = EraValidatorWeights::new(
+            block_1.header().era_id(),
+            BTreeMap::from([
+                (ALICE_PUBLIC_KEY.clone(), 10.into()),
```

Comment:
Please consider adding an explanation why these weights need to be different?

Reply:
Added in https://github.com/casper-network/casper-node/pull/3783/commits/5cb5b887afcefd75958bec58ffa70e0336246739